devices/tuya.js

```diff
@@ -4279,4 +4279,208 @@ module.exports = [
+        configure: async (device, coordinatorEndpoint, logger) => {
+            await tuya.configureMagicPacket(device, coordinatorEndpoint, logger);
+            await reporting.bind(device.getEndpoint(1), coordinatorEndpoint, ['genOnOff']);
+            await reporting.bind(device.getEndpoint(2), coordinatorEndpoint, ['genOnOff']);
+        },
+    },
+    {
+        fingerprint: [{
+            modelID: 'TS0003',
+            manufacturerName: '_TZ3000_pv4puuxi',
+        }],
+        model: 'TS0003_switch_3_gang',
+        vendor: 'TuYa',
+        description: '3-Gang switch with backlight',
+        fromZigbee: [tuya.fz.backlight_mode_off_on, fz.on_off, tuya.fz.power_on_behavior_2],
+        toZigbee: [tz.on_off, tuya.tz.power_on_behavior_2, tuya.tz.backlight_indicator_mode_2],
+        exposes: [
+            e.switch().withEndpoint('left'),
+            e.switch().withEndpoint('center'),
+            e.switch().withEndpoint('right'),
+            exposes.binary('backlight_mode', ea.ALL, 'on', 'off').withDescription('Backlight Mode'),
+        ],
+        endpoint: (device) => {
+            return {'left': 1, 'center': 2, 'right': 3};
+        },
+        meta: {multiEndpoint: true, disableDefaultResponse: true},
+        configure: async (device, coordinatorEndpoint, logger) => {
+            await tuya.configureMagicPacket(device, coordinatorEndpoint, logger);
+            await reporting.bind(device.getEndpoint(1), coordinatorEndpoint, ['genOnOff']);
+            await reporting.bind(device.getEndpoint(2), coordinatorEndpoint, ['genOnOff']);
+            await reporting.bind(device.getEndpoint(3), coordinatorEndpoint, ['genOnOff']);
+        },
+    },
+    {
+        fingerprint: [{
+            modelID: 'TS0601',
+            manufacturerName: '_TZE200_hewlydpz',
+        }],
+        model: 'TS0601_switch_4_gang_2',
+        vendor: 'TuYa',
+        description: '4-Gang switch with backlight',
+        fromZigbee: [tuya.fz.datapoints],
+        toZigbee: [tuya.tz.datapoints],
+        exposes: [
+            exposes.binary('switch_1', ea.STATE_SET, 'ON', 'OFF').withDescription('Switch 1'),
+            exposes.binary('switch_2', ea.STATE_SET, 'ON', 'OFF').withDescription('Switch 2'),
+            exposes.binary('switch_3', ea.STATE_SET, 'ON', 'OFF').withDescription('Switch 3'),
+            exposes.binary('switch_4', ea.STATE_SET, 'ON', 'OFF').withDescription('Switch 4'),
+            exposes.binary('backlight_mode', ea.STATE_SET, 'ON', 'OFF').withDescription('Backlight Mode'),
+            exposes.binary('child_lock', ea.STATE_SET, 'ON', 'OFF').withDescription('Child Lock'),
+            exposes.enum('relay_status', ea.STATE_SET, ['power_off', 'power_on', 'last']).withDescription('Relay Status'),
+        ],
+        configure: tuya.configureMagicPacket,
+        meta: {
+            tuyaDatapoints: [
+                [1, 'switch_1', tuya.valueConverter.onOffNotStrict],
```

Comment:
Please rename all `switch_X` to `state_lX`

Reply:
when using switch_l2 it expects endpoints for some reason, when using DP codes, there are no endpoints, it's general for the entire device, that's why I used switch_1

Reply:
to fix this, use the following in the definition:

```js
        meta: {
            multiEndpoint: true,
            tuyaDatapoints: [
                // ... all your datapoints here
            ],
        },
        endpoint: (device) => {
            return {'l1': 1, 'l2': 1, 'l2': 3, 'l2': 4};
        },
```

The advantage is that the exposes can be defined properly now, as:

```
exposes.binary('switch_1', ea.STATE_SET, 'ON', 'OFF').withDescription('Switch 1')
```


should be

```js
tuya.exposes.switch().withEndpoint('l1')
```

Reply:
I would prefer your approach as it much more readable and simple, will try it later this week and update, thanks

Reply:
just a questions about it, there are 2 different DP codes for each of the endpoints
101 - switch_l2
102 - switch_l1

how do I specify in that case which DP goes to which endpoint in tyaDatapoints?

thanks